elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchQueryProcessor.scala

```diff
@@ -128,8 +128,18 @@ class ElasticSearchQueryProcessor(val logicalPlan: LogicalPlan, val parameters:
   }
 
   private def selectFields(fields: Seq[Attribute], query: SearchDefinition): SearchDefinition = {
-      val stringFields: Seq[String] = fields.map(_.name)
-      query.fields(stringFields.toList: _*)
+      val subDocuments = schemaProvided.toSeq flatMap {
+        _.fields collect {
+          case StructField(name, _: StructType, _, _) => name
+        }
+      }
+      val stringFields: Seq[String] = fields.view map (_.name) filterNot (subDocuments contains _)
+
+      val fieldsQuery = query.fields(stringFields.toList: _*)
+
+      if(stringFields.size != fields.size)
```

Comment:
Unfortunately, that operation doesn't exists. Besides, imagine it does: It would be O(n Log n) at best, most probably O(n^2). Given that `stringFields` is the result of filtering out elements from `stringFields`, we know for sure that it is either empty or a sub set of the latter. Provided that condition you can check that one is different to the other by just comparing their size.

If, on the other hand, you take into consideration that most `Seq` implementers provide a O(1) length operation, not making use of the advantage of being able to compare them by just comparing their size would be a huge sacrifice of performance.

e.g: Blazing blazing fast size checking of huge streams:

```scala
val s = (1 to 1000000000).toSeq
s.size
```